java/ddlogapi/DDlogAPI.java

```diff
@@ -123,12 +123,14 @@ public DDlogAPI(int workers, Consumer<DDlogCommand> callback, boolean storeData)
     }
 
     /// Callback invoked from commit.
-    void onCommit(int tableid, long handle, boolean polarity) {
+    void onCommit(int tableid, long handle, long w) {
         if (this.commitCallback != null) {
-            DDlogCommand.Kind kind = polarity ? DDlogCommand.Kind.Insert : DDlogCommand.Kind.DeleteVal;
-            DDlogRecord record = DDlogRecord.fromSharedHandle(handle);
-            DDlogCommand command = new DDlogCommand(kind, tableid, record);
-            this.commitCallback.accept(command);
+            DDlogCommand.Kind kind = w > 0 ? DDlogCommand.Kind.Insert : DDlogCommand.Kind.DeleteVal;
+            for (long i = 0; i < java.lang.Math.abs(w); i++) {
```

Comment:
Do you intend this loop to execute only once when `w` is 1 or -1? If so I'd find an if statement to be more readable than a for loop.

Reply:
This is designed to work even without `consolidate` or `distinct`, in which case `w` can be an arbitrary value; hence the loop...